sycl/source/detail/queue_impl.cpp

```diff
@@ -97,9 +97,20 @@ event queue_impl::mem_advise(const shared_ptr_class<detail::queue_impl> &Self,
 }
 
 void queue_impl::addEvent(const event &Event) {
-  std::weak_ptr<event_impl> EventWeakPtr{getSyclObjImpl(Event)};
-  std::lock_guard<mutex_class> Lock(MMutex);
-  MEvents.push_back(std::move(EventWeakPtr));
+  // if the command behind the event has no memory dependencies,
+  // we need to track the event with the USMEvents, or it won't be properly
```

Comment:
I suggest renaming `MUSMEvents` & `MEvents` to something like `MEventsShared` and `MEventsWeak`, and specifying which events go where in their respective comments. The name `USMEvents` was a bit misleading before (it referred to events created by specifically queue's USM API) and it's even more misleading now (command groups with no dependencies do not necessarily use USM but still fall into this category as of this patch).

Reply:
I've done so. But I've left the function `queue_impl::addUSMEvent` named as such, to avoid an ABI change. If you (and @romanovvlad ) want that changed too, let me know.

Reply:
`addUSMEvent` is a private method and is not exported. Hence, changing its name won't brake ABI.

Reply:
`addUSMEvent` is not exported, but not because it is private or something, but rather because `queue_impl` is not marked with `__SYCL_EXPORT`.

Reply:
renamed